docs/source/Objective_Functions/Objective_Function.md

```diff
@@ -1,25 +1,203 @@
+}
+```
+
+#### The `Tell` Class 
+An example of `Tell` object values for an objective and local optimization function:
+```
+{
+    "parameters": [
+        {"values": [1, 3]},
+        {"values": [2, 7]}
+    ],
+    "headers": ["h0", "h1"],
+    "results": [6.59359884, 7.86938667],
+    #parameter candidates is only used for local optimization function
+    "candidates": [
+        {"values": [-8.18565e-09, -8.18565e-09]}
+    ]
+}
+```
+
+
+  An example of using an Ask and Tell object for receiving and passing the arguments to an Ackley function used as an objective function.
+
+```
+from tq42.channel import Ask, Tell
+import OptimizationTestFunctions as otf
+import numpy as np
+
+
+async def objective_function_callback(ask: Ask) -> Tell:     
+    dimensions = len(ask.headers)
+    
+    #initialize Ackley function to receive n dimension arrays
+    func = otf.Ackley(dimensions)
+    y = []
+    
+    #append the results of the Ackley function to y for all given n dimension parameters
+    for parameter in ask.parameters:
+        y.append(float(func(np.array(parameter.values))))
+
+    #create and return a Tell object which has the parameters and the results mapped together
+    tell = Tell(
+        parameters=ask.parameters,
+        headers=ask.headers,
+        results=y
+    )
+    return tell
+```
+
+An example of using an Ask and Tell object for receiving and passing the arguments to a local optimization function.
+
+Note these key points when constructing a Tell object for a local optimization function, 
+
+1. When creating a Tell object, you need an extra candidates parameter.
+2. Before adding results to the candidate list, map them to a string called "values".
+ 
+```
+
+async def local_optimization_function_callback(ask: Ask) -> Tell:
+    dim = len(ask.headers)
+    func = otf.Ackley(dim)
+    y = []
+    new_x = []
+    for parameter in ask.parameters:
+        res = optimize.minimize(func, np.array(parameter.values))
+        new_x.append({"values": res.x})
+        y.append(float(res.fun))
+
+    tell = Tell(
+        parameters=ask.parameters,
+        headers=ask.headers,
+        results=y,
+        candidates=new_x
+    )
+    return tell
+```
+
+We can then connect the channels created for `objective_func_channel` and `local_opt_channel`  to the objective function and local optimization function we created above using the connect API of the channel. The connect API connects to the stream and handles every message with the provided callback to create an answer.
+```
+The connect API parameters are:
+- callback: Async callback that handles an ASK message and returns a TELL message
+- finish_callback: Callback that is called when we finish the connection
+- int max_duration_in_sec: Timeout for whole connection in seconds. `None` -> no timeout for overall flow
+- int message_timeout_in_sec: Timeout between messages in seconds. Main way to end the connection.
+
+def success():
+    print("One async function done!")
```

Comment:
one done? Why not at least indicate which one is done :D?

Reply:
i modified the notebook to print the callback function name.